src/commands/Minion/groupkill.ts

```diff
@@ -39,28 +39,84 @@ export default class extends BotCommand {
 		return [quantity, duration, perKillTime];
 	}
 
-	checkReqs(users: KlasaUser[], monster: KillableMonster, quantity: number) {
+	async checkReqs(users: KlasaUser[], monster: KillableMonster, quantity: number) {
 		// Check if every user has the requirements for this monster.
+		const numberOfUsers = users.length;
+		let removedReasonStr: string[];
+		let foodId = 0;
+		let amountOfFoodNeeded = 0;
+
 		for (const user of users) {
+			removedReasonStr = [];
+			let checkReqsState = true;
 			if (!user.hasMinion) {
-				throw `${user.username} doesn't have a minion, so they can't join!`;
+				checkReqsState = false;
+				removedReasonStr.push(
+					`${user.username} doesn't have a minion, so they can't join!`
+				);
 			}
 
 			if (user.minionIsBusy) {
-				throw `${user.username} is busy right now and can't join!`;
+				checkReqsState = false;
+				removedReasonStr.push(`${user.username} is busy right now and can't join!`);
 			}
 
 			if (user.isIronman) {
-				throw `${user.username} is an ironman, so they can't join!`;
+				checkReqsState = false;
+				removedReasonStr.push(`${user.username} is an ironman, so they can't join!`);
 			}
 
 			const [hasReqs, reason] = user.hasMonsterRequirements(monster);
 			if (!hasReqs) {
-				throw `${user.username} doesn't have the requirements for this monster: ${reason}`;
+				checkReqsState = false;
+				removedReasonStr.push(
+					`${user.username} doesn't have the requirements for this monster: ${reason}`
+				);
 			}
 
-			if (1 > 2 && !hasEnoughFoodForMonster(monster, user, quantity)) {
-				throw `${user.username} doesn't have enough food.`;
+			if (1 > 2) {
+				let [healAmountNeeded] = calculateMonsterFood(monster, user);
+				healAmountNeeded = Math.ceil(healAmountNeeded / numberOfUsers);
+
+				for (const food of Eatables) {
+					const amountNeeded = ceil(healAmountNeeded / food.healAmount!) * quantity;
+					if (user.numItemsInBankSync(food.id) < amountNeeded) {
+						if (Eatables.indexOf(food) === Eatables.length - 1) {
+							checkReqsState = false;
+							removedReasonStr.push(
+								`${user.username} doesn't have enough food to kill ${
+									monster.name
+								} in this sized mass! You need enough food to heal atleast ${healAmountNeeded} HP (${healAmountNeeded /
+									quantity} per kill) You can use these food items: ${Eatables.map(
+									i => i.name
+								).join(', ')}.`
+							);
+						}
+
+						foodId = food.id;
```

Comment:
shouldnt these be one level higher, you dont care about the food that they dont have